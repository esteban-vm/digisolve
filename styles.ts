import { css } from '@emotion/react'

export const globalStyles = css`
  @import url('/css/fluid.css');
  @import url('/css/normalize.min.css');

  @font-face {
    font-family: 'Work Sans';
    src: local('Work Sans'), url('/fonts/WorkSans-VariableFont_wght.ttf');
  }

  :root {
    --color-primary: #ed07b7;
    --color-dark-primary: #bd0592;
    --color-secondary: #6920b0;
    --color-tertiary: #acdaff;

    --color-white: #fff;
    --color-grey: #555;
    --color-black: #000;

    --color-light: #f4f4f4;
    --color-lighter: #fcfcfc;

    --color-overlay1: #0000004d;
    --color-overlay2: #000000b3;
    --color-overlay3: #00000080;
    --color-overlay4: #000000cc;

    --color-shadow: #efefefef;
  }

  html,
  body {
    background-color: var(--color-white);
    color: var(--color-black);
    text-rendering: optimizeLegibility;
    font-size: 16px;
    font-family: 'Work Sans', Arial, sans-serif;
    font-weight: 300;
  }

  .row {
    max-width: 1140px;
    margin: 0 auto;
  }

  .clear_fix {
    zoom: 1;

    ::after {
      content: '';
      display: table;
      clear: both;
      visibility: hidden;
    }
  }
`

export * from '@emotion/react'
export { default as styled } from '@emotion/styled'
