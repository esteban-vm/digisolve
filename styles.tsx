import { Global, css } from '@emotion/react'

const styles = css`
  @import url('/css/fluid.css');
  @import url('/css/normalize.min.css');

  @font-face {
    font-family: 'Work Sans';
    src: local('Work Sans'), url('/fonts/WorkSans-VariableFont_wght.ttf');
  }

  :root {
    --primary-color: #ed07b7;
    --primary-color-dark: #bd0592;
    --secondary-color: #6920b0;
    --light-color: #acdaff;
    --dark-color: #555;
  }

  html,
  body {
    background-color: #fff;
    color: #000;
    text-rendering: optimizeLegibility;
    font-size: 16px;
    font-family: 'Work Sans', Arial, sans-serif;
    font-weight: 300;
  }

  .row {
    max-width: 1140px;
    margin: 0 auto;
  }
`

export const Globals = () => <Global styles={styles} />
export * from '@emotion/react'
export { default as styled } from '@emotion/styled'
