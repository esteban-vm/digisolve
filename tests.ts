import type { ReactElement } from 'react'
import { create, type ReactTestRendererJSON, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON() as ReactTestRendererJSON
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
