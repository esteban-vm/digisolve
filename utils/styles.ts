import { css } from '@emotion/react'

export enum breakPoints {
  /** Big tablets to smaller laptop screens */
  lg = '1200px',
  /** Smaller tablets to larger tablet screens */
  md = '1023px',
  /** Larger phones to smaller tablet screens */
  sm = '767px',
  /** Smaller to larger mobile phones */
  xs = '479px',
}

export const mediaQuery = (breakPoint: keyof typeof breakPoints) => {
  return `@media only screen and (max-width: ${breakPoints[breakPoint]})`
}

export const globalStyles = css`
  @import url('/css/fluid.min.css');
  @import url('/css/normalize.min.css');

  @font-face {
    font-family: 'Work Sans';
    src: local('Work Sans'), url('/fonts/WorkSans-VariableFont_wght.ttf');
  }

  :root {
    --color-primary: #ed07b7;
    --color-dark-primary: #bd0592;
    --color-secondary: #6920b0;
    --color-tertiary: #acdaff;
    --color-white: #fff;
    --color-grey: #555;
    --color-black: #000;
    --color-light: #f4f4f4;
    --color-lighter: #fcfcfc;
    --color-dark: #888;
    --color-darker: #333;
    --color-overlay1: #0000004d;
    --color-overlay2: #000000b3;
    --color-overlay3: #00000080;
    --color-overlay4: #000000cc;
    --color-shadow: #efefefef;
    --color-facebook: #3b5998;
    --color-twitter: #00aced;
    --color-google-plus: #dd4b39;
    --color-youtube: #f00;
    --color-instagram: #517fa4;
  }

  html,
  body {
    background-color: var(--color-white);
    color: var(--color-black);
    text-rendering: optimizeLegibility;
    font-size: 16px;
    font-family: 'Work Sans', Arial, sans-serif;
    font-weight: 300;

    ${mediaQuery('lg')} {
      font-size: 14px;
    }

    ${mediaQuery('md')} {
      font-size: 12px;
    }

    ${mediaQuery('sm')} {
      font-size: 10px;
    }
  }

  * {
    :focus {
      outline: none;
    }

    ::selection {
      background-color: var(--color-primary);
      color: var(--color-light);
    }

    ::-webkit-scrollbar {
      width: 0.5rem;

      &-track {
        box-shadow: inset 0 0 6px var(--color-darker);
      }

      &-thumb {
        background-color: var(--color-primary);
        outline: 1px solid var(--color-dark-primary);
        border-radius: 10px;
      }
    }
  }

  img,
  button,
  a {
    user-select: none;
  }

  ul {
    list-style: none;
  }

  .row {
    max-width: 1140px;
    margin: 0 auto;

    ${mediaQuery('md')} {
      padding: 0 2%;
    }
  }

  .clear_fix {
    zoom: 1;

    ::after {
      content: '';
      display: table;
      clear: both;
      visibility: hidden;
    }
  }

  .section {
    &__heading {
      font-size: 250%;
      text-transform: capitalize;
      color: var(--color-black);
      margin-bottom: 20px;
      margin-top: 150px;

      ${mediaQuery('lg')} {
        margin-top: 90px;
      }

      ${mediaQuery('sm')} {
        margin-top: 60px;
        margin-bottom: 10px;
      }
    }

    &__subheading {
      font-size: 150%;
      color: var(--color-grey);
      font-weight: 300;
      margin-bottom: 30px;
      line-height: 1.5;

      ::after {
        content: ' ';
        display: block;
        height: 2px;
        width: 100px;
        background-color: var(--color-primary);
        margin: 30px auto 0;

        ${mediaQuery('sm')} {
          margin-top: 15px;
        }
      }
    }
  }
`

export * from '@emotion/react'
export { default as styled } from '@emotion/styled'
