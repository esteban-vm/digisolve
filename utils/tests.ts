import type { ReactElement } from 'react'
import { create, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON()
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
