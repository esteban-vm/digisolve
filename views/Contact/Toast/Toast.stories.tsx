import type { Meta } from '@/stories'
import Toast from './Toast'

export default {
  component: Toast,
  title: 'Views/Contact/Toast',
  parameters: {
    layout: 'centered',
    viewport: {
      disable: true,
    },
  },
} as Meta<typeof Toast>

export const Example = () => {
  return (
    <Toast isOpen>
      <p>Test</p>
    </Toast>
  )
}
