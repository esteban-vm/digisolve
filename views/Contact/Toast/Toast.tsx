import type { Component, PropsWithChildren } from '@/types'
import { useRef, useEffect } from 'react'
import { Button, Icon } from '@/components'
import { styled, mediaQuery } from '@/styles'

type ToastProps = Required<PropsWithChildren<{ isOpen: boolean }>>

const ToastComponent: Component<ToastProps> = ({ isOpen, children, ...rest }) => {
  const toastRef = useRef<HTMLDialogElement>(null)

  useEffect(() => {
    if (isOpen) toastRef.current?.showModal()
  }, [isOpen])

  const closeToast = () => {
    toastRef.current?.close()
  }

  return (
    <dialog ref={toastRef} {...rest}>
      <div className='content'>
        <h2>Submitted successfully</h2>
        <Icon
          name='envelope-circle-check'
          size={3}
          className='animate__animated animate__heartBeat animate__infinite icon'
        />
        {children}
        <Button text='ok' isClose onClose={closeToast} className='btn' />
      </div>
    </dialog>
  )
}

const Toast = styled(ToastComponent)`
  text-align: center;
  color: var(--color-secondary);
  background-color: var(--color-light);
  border: 2px solid var(--color-primary);
  border-radius: 5px;
  max-width: 60%;

  ${mediaQuery('xs')} {
    max-width: 90%;
  }

  h2 {
    margin: 0;
    color: var(--color-primary);
  }

  .content {
    padding: 2rem;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }

  .icon {
    color: inherit;
    margin: 10px auto;
  }

  .btn {
    margin-top: 10px;
  }
`

export default Toast
