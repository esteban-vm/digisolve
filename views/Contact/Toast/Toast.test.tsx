import { render, cleanup, screen, create, axe } from '@/tests'
import { mediaQuery } from '@/styles'
import Toast from './Toast'

describe('ðŸ§ª TOAST:', () => {
  describe('render tests:', () => {
    let container: HTMLElement

    beforeEach(() => {
      void ({ container } = render(
        <Toast isOpen>
          <p>Test</p>
        </Toast>
      ))
    })

    afterEach(cleanup)

    it('should be accessible', async () => {
      const results = await axe(container)
      expect(results).toHaveNoViolations()
    })

    describe('should be rendered:', () => {
      it('the title', () => {
        const title = screen.getByRole('heading', {
          name: /^submitted successfully$/i,
          level: 2,
          hidden: true,
        })

        expect(title).toBeInTheDocument()
      })

      it('the icon', () => {
        const icon = screen.getByRole('img', { hidden: true })
        expect(icon).toBeInTheDocument()
      })

      it('the button', () => {
        const button = screen.getByRole('button', { name: /^ok$/i, hidden: true })
        expect(button).toBeInTheDocument()
      })

      it('children elements', () => {
        const paragraph = screen.getByText(/^test$/i)
        expect(paragraph).toBeInTheDocument()
      })
    })
  })

  describe('style tests:', () => {
    let tree: ReturnType<typeof create>

    beforeEach(() => {
      tree = create(
        <Toast isOpen>
          <p>Test</p>
        </Toast>
      )
    })

    it('should render with correct styles', () => {
      expect(tree).toMatchSnapshot()
    })

    it('should have correct styles in backdrop pseudo-element', () => {
      const target = '::backdrop'
      expect(tree).toHaveStyleRule('background-color', 'black', { target })
      expect(tree).toHaveStyleRule('opacity', '0.5', { target })
    })

    it('should have correct styles on extra small screen devices', () => {
      expect(tree).toHaveStyleRule('max-width', '90%', { media: mediaQuery('xs') })
    })
  })
})
