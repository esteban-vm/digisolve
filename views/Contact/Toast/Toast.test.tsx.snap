// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª TOAST: style tests: should render with correct styles 1`] = `
.emotion-0 {
  text-align: center;
  color: var(--color-secondary);
  background-color: var(--color-light);
  border: 2px solid var(--color-primary);
  border-radius: 5px;
  max-width: 60%;
}

@media only screen and (max-width: 480px) {
  .emotion-0 {
    max-width: 90%;
  }
}

.emotion-0 h2 {
  margin: 0;
  color: var(--color-primary);
}

.emotion-0 .content {
  padding: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 .icon {
  color: inherit;
  margin: 10px auto;
}

.emotion-0 .btn {
  margin-top: 10px;
}

.emotion-1 {
  cursor: pointer;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: capitalize;
  padding: 10px 30px;
  border-radius: 200px;
  font-weight: 500;
  outline: none;
  -webkit-transition: background-color 500ms;
  transition: background-color 500ms;
  border: 2px solid var(--color-primary);
  background-color: transparent;
  color: var(--color-primary);
}

.emotion-1:hover {
  background-color: var(--color-primary);
  color: var(--color-white);
}

<dialog
  className="emotion-0"
>
  <div
    className="content"
  >
    <h2>
      Submitted successfully
    </h2>
    <svg
      aria-hidden={true}
      className="animate__animated animate__heartBeat animate__infinite icon"
      color="var(--color-primary)"
      fill="currentColor"
      height="3rem"
      role="img"
      viewBox="0 0 640 512"
      width="3rem"
    >
      <path
        d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4l217.6 163.2c11.4 8.5 27 8.5 38.4 0l57.4-43c23.9-59.8 79.7-103.3 146.3-109.8l13.9-10.4c12.1-9.1 19.2-23.3 19.2-38.4 0-26.5-21.5-48-48-48H48zm246.4 275.2a63.9 63.9 0 01-76.8 0L0 176v208c0 35.3 28.7 64 64 64h296.2c-25.1-30.4-40.2-69.5-40.2-112 0-5.6.3-11.1.8-16.6l-26.4 19.8zM640 336c0-79.5-64.5-144-144-144s-144 64.5-144 144 64.5 144 144 144 144-64.5 144-144zm-76.7-43.3c6.2 6.2 6.2 16.4 0 22.6l-72 72c-6.2 6.2-16.4 6.2-22.6 0l-40-40c-6.2-6.2-6.2-16.4 0-22.6s16.4-6.2 22.6 0l28.7 28.7 60.7-60.7c6.2-6.2 16.4-6.2 22.6 0z"
      />
    </svg>
    <p>
      Test
    </p>
    <button
      className="btn emotion-1"
      onClick={[Function]}
      type="button"
    >
      ok
    </button>
  </div>
</dialog>
`;
